src/main/java/com/google/sps/servlets/joinGameServlet.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+ 
+package com.google.sps.servlets;
+ 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.utils.GameUtils;
+import com.google.sps.utils.UserUtils; 
+import java.io.*;
+import javax.servlet.*;
+import javax.servlet.http.*;
+import java.io.PrintWriter;
+ 
+/** When a user joins a game, check if the given game id exists
+    if so, get the game entity and add the user,
+    then add the game to the user entity.
+    If it doesn't exist, redirect back to the join game page.
+*/
+@WebServlet("/join-game")
+public class joinGameServlet extends HttpServlet {
+ 
+  DatastoreService datastore;
+ 
+  public joinGameServlet(){
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+ 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    String gameId = request.getParameter("game-id");
+ 
+    // get user entity
+    Cookie cookies[] = request.getCookies();
+    Entity userEntity = UserUtils.getUserFromCookie(cookies, datastore);
+
+    String joinGame = GameUtils.joinGame(userEntity, null, datastore, gameId);
```

Comment:
I just left some comments on https://github.com/googleinterns/step42-2020/pull/40 regarding `GameUtils.joinGame`. This PR could be simpler if `GameUtils.joinGame` were simplified in the ways I suggested.